connection.go

```diff
@@ -547,6 +570,155 @@ func (sc *snowflakeConn) getQueryResult(ctx context.Context, resultPath string)
 	return respd, nil
 }
 
+func isAsyncMode(ctx context.Context) (bool, error) {
+	val := ctx.Value(AsyncMode)
+	if val == nil {
+		return false, nil
+	}
+	boolVal, ok := val.(bool)
+	if !ok {
+		return false, fmt.Errorf("failed to cast val %+v to bool", val)
+	}
+	return boolVal, nil
+}
+
+func getAsync(
+	ctx context.Context,
+	sr *snowflakeRestful,
+	headers map[string]string,
+	URL *url.URL,
+	timeout time.Duration,
+	res *snowflakeResult,
+	rows *snowflakeRows,
+) {
+	resType, _ := ctx.Value(resultType).(string)
+	headers[headerAuthorizationKey] = fmt.Sprintf(headerSnowflakeToken, sr.Token)
+	resp, err := sr.FuncGet(ctx, sr, URL, headers, timeout)
+	if err != nil {
+		logger.WithContext(ctx).Errorf("failed to get response. err: %v", err)
+		if resType == "exec" {
+			res.statusChannel <- queryEvent{
+				QueryFailed,
+				&SnowflakeError{
+					Number:  -1,
+					Message: err.Error(),
+					QueryID: res.queryID,
+				},
+			}
+			close(res.statusChannel)
+		} else {
+			rows.statusChannel <- queryEvent{
+				QueryFailed,
+				&SnowflakeError{
+					Number:  -1,
+					Message: err.Error(),
+					QueryID: rows.queryID,
+				},
+			}
+			close(rows.statusChannel)
+		}
+		return
+	}
+
+	respd := execResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&respd)
+	resp.Body.Close()
+	if err != nil {
+		logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
+		if resType, _ := ctx.Value(resultType).(string); resType == "exec" {
+			res.statusChannel <- queryEvent{
+				QueryFailed,
+				&SnowflakeError{
+					Number:  -1,
+					Message: err.Error(),
+					QueryID: res.queryID,
+				},
+			}
+			close(res.statusChannel)
+		} else {
+			rows.statusChannel <- queryEvent{
+				QueryFailed,
+				&SnowflakeError{
+					Number:  -1,
+					Message: err.Error(),
+					QueryID: rows.queryID,
+				},
+			}
+			close(rows.statusChannel)
+		}
+		return
+	}
+
+	var code int
+	if respd.Code != "" {
+		code, err = strconv.Atoi(respd.Code)
+		if err != nil {
+			code = -1
+		}
+	} else {
+		code = -1
+	}
+
+	if resType, _ := ctx.Value(resultType).(string); resType == "exec" {
```

Comment:
Do we need to do this casting every time? Can it be done at the top of the func?